compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformInsets.skiko.kt

```diff
@@ -22,117 +22,44 @@ import androidx.compose.runtime.Stable
-        if (this === other) return true
-        if (other !is PlatformInsets) return false
-
-        if (left != other.left) return false
-        if (top != other.top) return false
-        if (right != other.right) return false
-        if (bottom != other.bottom) return false
-
-        return true
-    }
-
-    override fun hashCode(): Int {
-        var result = left.hashCode()
-        result = 31 * result + top.hashCode()
-        result = 31 * result + right.hashCode()
-        result = 31 * result + bottom.hashCode()
-        return result
-    }
-
-    override fun toString(): String {
-        return "PlatformInsets(left=$left, top=$top, right=$right, bottom=$bottom)"
-    }
-}
-
-internal fun PlatformInsets.union(insets: PlatformInsets) = PlatformInsets(
-    left = maxOf(left, insets.left),
-    top = maxOf(top, insets.top),
-    right = maxOf(right, insets.right),
-    bottom = maxOf(bottom, insets.bottom)
-)
-
-internal fun PlatformInsets.exclude(insets: PlatformInsets) = PlatformInsets(
-    left = (left - insets.left).coerceAtLeast(0.dp),
-    top = (top - insets.top).coerceAtLeast(0.dp),
-    right = (right - insets.right).coerceAtLeast(0.dp),
-    bottom = (bottom - insets.bottom).coerceAtLeast(0.dp)
-)
-
-internal fun lerp(start: PlatformInsets, stop: PlatformInsets, fraction: Float) =
-    PlatformInsets(
-        left = lerp(start.left, stop.left, fraction),
-        right = lerp(start.right, stop.right, fraction),
-        top = lerp(start.top, stop.top, fraction),
-        bottom = lerp(start.bottom, stop.bottom, fraction)
-    )
+interface PlatformInsets {
+    /**
+     * The left inset in pixels.
+     */
+    val left: Int
+
+    /**
+     * The top inset in pixels.
+     */
+    val top: Int
+
+    /**
+     * The right inset in pixels.
+     */
+    val right: Int
+
+    /**
+     * The bottom inset in pixels.
+     */
+    val bottom: Int
 
-internal interface InsetsConfig {
-
-    // TODO: Add more granular control. Look at Android's [WindowInsetsCompat]
-    val safeInsets: PlatformInsets
-        @Composable get
-
-    val ime: PlatformInsets
-        @Composable get
-
-    // Don't make it public, it should be implementation details for creating new root layout nodes.
-    // TODO: Ensure encapsulation and proper control flow during refactoring [Owner]s
-    @Composable
-    fun excludeInsets(
-        safeInsets: Boolean,
-        ime: Boolean,
-        content: @Composable () -> Unit
-    )
-}
-
-internal object ZeroInsetsConfig : InsetsConfig {
-    override val safeInsets: PlatformInsets
-        @Composable get() = PlatformInsets.Zero
-
-    override val ime: PlatformInsets
-        @Composable get() = PlatformInsets.Zero
-
-    @Composable
-    override fun excludeInsets(
-        safeInsets: Boolean,
-        ime: Boolean,
-        content: @Composable () -> Unit
-    ) {
-        content()
+    companion object {
+        val Unspecified = object : PlatformInsets {
```

Comment:
Please define `ValuePlatformInsets` instead of producing multiple classes with const values. This type might be internal